src/daft-session/src/session.rs

```diff
@@ -263,6 +283,13 @@ impl SessionState {
             _ => panic!("ambiguous table identifier"),
         }
     }
+
+    pub fn get_function(&self, name: &str) -> Result<WrappedUDFClass> {
+        self.functions
+            .get_exact(name)
+            .cloned()
+            .ok_or_else(|| crate::error::Error::function_not_found(name))
```

Comment:
Functions in SQL are subject to the case-normalization / case-lookup rules so will need to use the lookup method and error on ambiguity in the case of multiple matches.